csrc/execution/type_traits.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+// Modified from
+// https://github.com/facebookexperimental/libunifex/blob/main/include/unifex/type_traits.hpp
+
+#ifndef MMDEPLOY_CSRC_EXPERIMENTAL_EXECUTION_TYPE_TRAITS_H_
+#define MMDEPLOY_CSRC_EXPERIMENTAL_EXECUTION_TYPE_TRAITS_H_
+
+#include <type_traits>
+
+namespace mmdeploy {
+
+/////////////////////////////////////////////////////////
+// remove_cvref without handling volatile
+template <typename T>
+struct remove_cvref {
+  using type = T;
+};
+template <typename T>
+struct remove_cvref<const T> {
+  using type = T;
+};
+template <typename T>
+struct remove_cvref<T&> {
+  using type = T;
+};
+template <typename T>
+struct remove_cvref<const T&> {
+  using type = T;
+};
+template <typename T>
+struct remove_cvref<T&&> {
+  using type = T;
+};
+template <typename T>
+struct remove_cvref<const T&&> {
+  using type = T;
+};
+
+template <typename T>
+using remove_cvref_t = typename remove_cvref<T>::type;
+
+template <typename Fn, typename... Args>
+using callable_result_t = decltype(std::declval<Fn&&>()(std::declval<Args&&>()...));
+
+namespace _is_callable {
```

Comment:
It seem that `is_callable` and `defer` trait has not been used. Did you plan to use it in `tag_invoke`?